runhouse/servers/cluster_servlet.py

```diff
@@ -258,46 +238,43 @@ async def aperiodic_cluster_checks(self):
                     break
 
                 logger.debug("Performing cluster checks")
-                status: ResourceStatusData = await self.astatus()
+                status, den_resp = await self.astatus(
+                    send_to_den=should_send_status_and_logs_to_den
+                )
+
                 if should_update_autostop:
                     logger.debug("Updating autostop")
                     await self._update_autostop(status)
 
                 if not should_send_status_and_logs_to_den:
                     break
 
-                logger.debug("Sending cluster status to Den")
-                cluster_rns_address = cluster_config.get("name")
-                cluster_uri = rns_client.format_rns_address(cluster_rns_address)
-                api_server_url = status.cluster_config.get(
-                    "api_server_url", rns_client.api_server_url
-                )
-
-                resp = await ClusterServlet.save_status_metrics_to_den(
-                    status=status,
-                    cluster_uri=cluster_uri,
-                    api_server_url=api_server_url,
-                )
-                status_code = resp.status_code
+                status_code = den_resp.status_code
 
                 if status_code == 404:
                     logger.info(
                         "Cluster has not yet been saved to Den, cannot update status or logs."
                     )
                 elif status_code != 200:
                     logger.error(
-                        f"{status_code}: Failed to send cluster status to Den: {resp.json()}"
+                        f"Failed to send cluster status to Den: {den_resp.json()}"
                     )
                 else:
```

Comment:
is it not possible for this to be some other error code? what if theres a 500? we would still be treating it as a successful request here